nemo/collections/asr/modules/conv_asr.py

```diff
@@ -463,6 +469,31 @@ def _prepare_for_export(self, **kwargs):
             logging.warning(f"Turned off {m_count} masked convolutions")
         Exportable._prepare_for_export(self, **kwargs)
 
+    # Adapter method overrides
+    def add_adapter(self, name: str, cfg: DictConfig):
+        # Update the config with correct input dim
+        cfg = self._update_adapter_cfg_input_dim(cfg)
+        # Add the adapter
+        super().add_adapter(name=name, cfg=cfg)
+
+    def _update_adapter_cfg_input_dim(self, cfg: DictConfig):
+        if 'in_features' in cfg:
+            in_planes = cfg['in_features']
+
+            if in_planes != self._feat_in:
+                logging.info(
+                    f"Updating {self.__class__.__name__} Adapter input dim from {in_planes} " f"to {self._feat_in}"
+                )
+                in_planes = self._feat_in
+
+            cfg['in_features'] = in_planes
+            return cfg
+        else:
+            raise ValueError(
```

Comment:
What are the cases where in_features is missing in the config? cases like user forgets to have it?

Reply:
In case users forget to add it, or if they use an adapter module which doesn't have the keyword arg `in_features`. For example, a user created adapter module that does not use in_features but input_dim as name. It is a little brittle, but covers the most common case.

Reply:
I will simplify this method.